plugins/uefi/fu-plugin-uefi.c

```diff
@@ -701,42 +633,41 @@ fu_plugin_uefi_delete_old_efivars (FuPlugin *plugin, GError **error)
 }
 
 /* remove when https://github.com/rhboot/efivar/pull/100 merged */
-static int
-_efi_get_variable_exists (efi_guid_t guid, const char *name)
+static gint
+_efi_get_variable_exists (efi_guid_t guid, const gchar *name)
 {
-	uint32_t unused_attrs = 0;
+	guint32 unused_attrs = 0;
 	return efi_get_variable_attributes (guid, name, &unused_attrs);
 }
 
 gboolean
 fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
+	const gchar *key = "ESPMountPoint";
 
-	/* get the supported status */
-	data->esrt_status = fwup_supported ();
-	if (data->esrt_status == FWUP_SUPPORTED_STATUS_UNSUPPORTED) {
-		g_set_error_literal (error,
+	/* load from file */
+	data->esp_path = fu_plugin_get_config_value (plugin, key);
+	if (data->esp_path != NULL) {
+		if (!g_file_test (data->esp_path, G_FILE_TEST_IS_DIR)) {
+			g_set_error (error,
 				     FWUPD_ERROR,
-				     FWUPD_ERROR_NOT_SUPPORTED,
-				     "UEFI firmware updating not supported");
-		return FALSE;
-	}
-
-	/* load any overriden options */
-	if (!fu_plugin_uefi_set_custom_mountpoint (plugin, error))
-		return FALSE;
+				     FWUPD_ERROR_INVALID_FILE,
+				     "Invalid %s specified in %s config: %s",
+				     fu_plugin_get_name (plugin), key,
+				     data->esp_path);
 
-	/* get the default compiled-in value for the ESP mountpoint */
-#ifdef HAVE_FWUP_GET_ESP_MOUNTPOINT
-	if (data->esp_path == NULL)
-		data->esp_path = g_strdup (fwup_get_esp_mountpoint ());
-#endif
+			return FALSE;
+		}
+	}
 
 	/* fall back to a sane default */
 	if (data->esp_path == NULL)
 		data->esp_path = g_strdup ("/boot/efi");
 
+	//FIXME: hardcoded?
+	data->esrt_path = g_strdup ("/sys/firmware/efi/esrt");
+
```

Comment:
Yeah I don't see a problem with hardcoding this path.  Even if mock is used for tests it will mock up this directory for you.

Reply:
I wondered about using fu_common_get_path() and then we can trivially test it using a static tree in data/tests...